lib/livebook/settings.ex

```diff
@@ -117,4 +130,113 @@ defmodule Livebook.Settings do
   def set_update_check_enabled(enabled) do
     storage().insert(:settings, "global", update_check_enabled: enabled)
   end
+
+  @doc """
+  Gets a list of environment variables from storage.
+  """
+  @spec fetch_env_vars() :: list(EnvVar.t())
+  def fetch_env_vars do
+    for fields <- storage().all(:env_vars) do
+      struct(EnvVar, fields)
+    end
+  end
+
+  @doc """
+  Gets one environment variable from storage.
+
+  Raises `RuntimeError` if the environment variable does not exist.
+  """
+  @spec fetch_env_var!(String.t()) :: EnvVar.t()
+  def fetch_env_var!(id) do
+    case storage().fetch(:env_vars, id) do
+      :error ->
+        raise NotFoundError,
+          message: "could not find an environment variable matching #{inspect(id)}"
+
+      {:ok, fields} ->
+        struct(EnvVar, fields)
+    end
+  end
+
+  @doc """
+  Checks if environment variable already exists.
+  """
+  @spec env_var_exists?(String.t()) :: boolean()
+  def env_var_exists?(id) do
+    storage().fetch(:env_vars, id) != :error
+  end
+
+  @doc """
+  Persists an environment variable from given changeset.
```

Comment:
this function accepts the struct, not the changeset.

```suggestion
  Persists the given environment variable.
```